### PR TITLE
improvements

### LINKED ISSUE #257: Remove Apache commons dependency

### PR DESCRIPTION
**Pull request by metacosm**
- refactor: use java.util.Set operation instead of Sets from Arc
- refactor: use String.join instead of stream operations
- refactor: rename getYAMLData to more appropriate getManifestData
- refactor: remove unneeded field
- refactor: use append instead of concatenation
- fix: remove unused warnings
- fix: wrong field was used
- refactor: minor improvements
- refactor: remove dependency on commons-lang3 Fixes #257
- fix: logging

### CODE REVIEW

File: samples/mysql-schema/pom.xml [hunk 1]
```diff
@@ -30,10 +30,6 @@
       <groupId>io.quarkus</groupId>
       <artifactId>quarkus-container-image-jib</artifactId>
     </dependency>
-    <dependency>
```
Comment (scrocquesel) [inline samples/mysql-schema/pom.xml:33]: How can it compile when it is still used ?

https://github.com/quarkiverse/quarkus-operator-sdk/blob/1d7c65a9c70a251ed84f66e869ccbc2d3a0f46ca/samples/mysql-schema/src/main/java/io/quarkiverse/operatorsdk/samples/mysqlschema/MySQLSchemaReconciler.java#L11

Comment (metacosm) [reply samples/mysql-schema/pom.xml:33]: Hmm, actually, I made the wrong change, I was planning on leaving the samples alone. 😅
That said the dependency is probably coming transitively…

Comment (scrocquesel) [reply samples/mysql-schema/pom.xml:33]: yes it i brought by io.quarkus:quarkus-core-deployment.
+1 for letting it in the sample :)

File: bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/builders/CsvManifestsBuilder.java [hunk 2]
```diff
@@ -162,13 +161,10 @@ private void handlePermissions(List<ClusterRole> clusterRoles, List<RoleBinding>
         Map<String, List<PolicyRule>> customPermissionRules = new HashMap<>();
         if (metadata.permissionRules != null) {
             for (CSVMetadataHolder.PermissionRule permissionRule : metadata.permissionRules) {
-                String serviceAccountName = StringUtils.defaultIfEmpty(permissionRule.serviceAccountName,
+                String serviceAccountName = defaultIfEmpty(permissionRule.serviceAccountName,
```
Comment (Sgitario) [inline bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/builders/CsvManifestsBuilder.java:164]: Can this line be written in a single line now?

**Review by Sgitario — APPROVED** [review]

File: samples/mysql-schema/pom.xml (hunk 1 above)
Comment (metacosm) [reply samples/mysql-schema/pom.xml:33]: By the way, I'm having issues with the MySQL test locally: it times out here. I'm wondering if there's an issue with Docker and testcontainers on M1 macs…

File: bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/builders/CsvManifestsBuilder.java (hunk 2 above)
Comment (metacosm) [reply bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/builders/CsvManifestsBuilder.java]: Done.

File: samples/mysql-schema/pom.xml (hunk 1 above)
Comment (scrocquesel) [reply samples/mysql-schema/pom.xml:33]: > By the way, I'm having issues with the MySQL test locally: it times out here. I'm wondering if there's an issue with Docker and testcontainers on M1 macs…

@metacosm On some environnement (especially when the docker daemon is not on the host but in a wrapper VM like with Docker Desktop for win/mac), exposed containers are not accessible on localhost and you need to set an env var `TESTCONTAINERS_HOST_OVERRIDE=host.docker.internal` with the correct hostname.

**Pull request merged**